qmd/sets.qmd

```diff
@@ -15,12 +15,28 @@ Sets are lists of categories and other parts. For example, `Collection category
 
 Sets can have parts that are reused. For example, there are different quality measures for sample collection (e.g. `Sample not collected`), PCR measures (`Inhibition preset`), and sequencing measures (`Low depth of coverage`), but all quality sets share the `No quality concerns` indicator.
 
-```{r, echo=FALSE, warning=FALSE}
+```{r, echo=FALSE, warning=TRUE}
 # Utility Functions
 # Source utility functions
 source("R/qmd-utils.R")
 # Source table formatting
 source("R/format-table.R")
+# Source warning utils
+source("R/warning-utils.R")
+source("R/template-utils.R")
+
+# Display string templates
+header_template <-
+  '## {partLabel} {{ #{partID} }}
+  <a href=\"/parts.html#{partID}\">({partID})</a>. {partDesc} {if (partInstr %!=na% constants$dictionary_missing_value_replacement) partInstr else "" }'
+
+set_sub_header_template <- '<a href=\"/parts.html#{partID}\">({partID})</a>{partDesc} \\
+{if(partInstr %!=na% constants$dictionary_missing_value_replacement) partInstr else ""} </br>
+Status: {status}. First Released: {firstReleased}. \\
+Last updated: {lastUpdated}.'
+
+part_bullet_template <- '**{partLabel}** <a href=\"/parts.html#{partID}\">({partID})</a>. {partDesc} </br>Status: {status}. First released: {firstReleased}. Last updated: {lastUpdated}.'
```

Comment:
This template is used three times in the code and it looks like the code to get the display inputs is the same. Can we pull it out into a function? Specifically these lines:
1. https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/warning-refactor/qmd/sets.qmd#L165-L176
2. https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/warning-refactor/qmd/sets.qmd#L233-L244
3. https://github.com/Big-Life-Lab/PHES-ODM-Doc/blob/warning-refactor/qmd/sets.qmd#L288-L299